demisto_sdk/commands/download/downloader.py

```diff
@@ -750,520 +863,434 @@ def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
-        """
-        Returns the corresponding pack content object to the given custom content object
-        :param custom_content_object: The custom content object to merge into the pack
-        :return: The corresponding pack content object
-        """
-        file_entity: str = custom_content_object["entity"]
-        file_name: str = custom_content_object["name"]
-        pack_entity_instances: list = self.pack_content[file_entity]
+        shutil.move(src=file_path, dst=content_item_output_path)
 
-        for pack_entity_instance in pack_entity_instances:
-            if file_name in pack_entity_instance:
-                return pack_entity_instance
-        return {}
+        try:  # Clean up temp dir
+            shutil.rmtree(temp_dir, ignore_errors=True)
+        
+        except shutil.Error as e:
+            logger.warning(f"Failed to remove temp dir '{temp_dir}': {e}")
+            logger.debug(traceback.format_exc())
+
+            if self.run_format and content_item_output_path.suffix in (".yml", ".yaml", ".json"):
+                format_manager(
+                    input=str(content_item_output_path),
+                    no_validate=False,
+                    assume_answer=False,
+                )
 
-    @staticmethod
-    def get_corresponding_pack_file_object(
-        searched_basename: str, pack_content_object: dict
-    ) -> dict:
-        """
-        Searches for the file named searched_basename under the pack content object and returns it
-        :param searched_basename: The basename to look for
-        :param pack_content_object: The pack content object
-        :return: The pack file object
-        """
-        for _, file_objects in pack_content_object.items():
-            for file_object in file_objects:
-                if file_object["path"].endswith(searched_basename):
-                    return file_object
-        return {}
+        logger.debug(f"Content item '{content_item_name}' was successfully downloaded.")
+        return True
 
     @staticmethod
-    def update_data(output_path: str, file_path_to_read: str, file_ending: str) -> None:
+    def update_data(file_to_update: Path | str, original_file: str, is_yaml: bool) -> None:
         """
         Collects special chosen fields from the file_path_to_read and writes them into the file_path_to_write.
-        :param file_path_to_write: The output file path to add the special fields to.
-        :param file_path_to_read: The input file path to read the special fields from.
-        :param file_ending: The files ending
-        :return: None
+
+        Args:
+            file_to_update (Path | str): Path to the new file to merge 'original_file' into.
+            original_file (str): Path to the original file to merge into 'file_to_update'.
+            is_yaml (bool): True if the file is a yml file, False if it's a json file.
         """
+        file_to_update = Path(file_to_update) if isinstance(file_to_update, str) else file_to_update
```

Comment:
```suggestion
        file_to_update = Path(file_to_update)
```
Path(Path("a")) == Path("a")